lib/mapper.js

```diff
@@ -22,13 +52,52 @@ exports.track = function(track){
     objCase.del(props, 'revenue');
   }
 
-  return {
-    email: track.email(),
-    userId: track.userId(),
-    eventName: track.event(),
-    createdAt: unixTime(track.timestamp()),
-    dataFields: props
-  };
+  var eventName = track.event();
+  var cart = props["products"] || props["cart"];
```

Comment:
The products array should be accessed with the proper getter method on the track facade object (`track.products()`). Then you can wrap each product like so:

```js
var Track = require('segmentio-facade').Track;
var products = track.products().map(function(item){
    var product = new Track({ properties: item });
    return product;
}
```

then you can use array methods on the products and use proper facade getters on each product for proper casing insensitivity and sensible defaults, etc.

Great example case: https://github.com/segmentio/integration-klaviyo/blob/master/lib/mapper.js#L60